File: backend/ngts/src/main/java/com/three/ngts/Entity/Driver.java
package com.three.ngts.Entity;

import jakarta.persistence.*;
import lombok.Data;

import java.sql.Date;

@Data
@Entity
@Table(name = "driver")
public class Driver {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "driver_id")
    private Long driverId;

    @Column(name = "driver_license_no")
    private String driverLicenseNo;

    @Column(name = "driver_license_exp")
    private Date driverLicenseExp;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User userId;
}

File: frontend/next-generation-ticketing-system/src/Entity/Driver.ts
class Driver {
  driverId: number;
  driverLicenseNo: string;
  driverLicenseExp: Date;
  userId: number;

  constructor(
    driverId: number,
    driverLicenseNo: string,
    driverLicenseExp: Date,
    userId: number
  ) {
    this.driverId = driverId;
    this.driverLicenseNo = driverLicenseNo;
    this.driverLicenseExp = driverLicenseExp;
    this.userId = userId;
  }
}

export default Driver;
